bedrock/firefox/views.py

```diff
@@ -242,6 +242,106 @@ def send_to_device_ajax(request):
     return HttpResponseJSON(resp_data)
 
 
+def firefox_all(request, platform, channel):
+    locale = l10n_utils.get_locale(request)
+
+    # Only show the new template at /firefox/all/ until enough locales have it translated.
+    # Once we no longer need the old template we can redirect the old URLs to the unified page.
+    if platform is None and channel is None and lang_file_is_active('firefox/all-unified', locale):
+        view = all_downloads_unified(request)
+    else:
+        view = all_downloads(request, platform, channel)
+
+    return view
+
+
+def all_downloads_unified(request):
+    product_android = firefox_android
+    product_desktop = firefox_desktop
+
+    # Human-readable product labels
+    products = OrderedDict([
+        ('desktop_release', 'Firefox'),
+        ('desktop_beta', 'Firefox Beta'),
+        ('desktop_developer', 'Firefox Developer Edition'),
+        ('desktop_nightly', 'Firefox Nightly'),
+        ('desktop_esr', 'Firefox Extended Support Release'),
+        ('android_release', 'Firefox Android'),
+        ('android_beta', 'Firefox Android Beta'),
+        ('android_nightly', 'Firefox Android Nightly'),
+    ])
+
+    channel_release = 'release'
+    channel_beta = 'beta'
+    channel_alpha = 'alpha'
+    channel_nightly = 'nightly'
+    channel_esr = 'esr'
+    channel_esr_next = 'esr_next'
```

Comment:
@pmac once thing I had some difficulty testing was how to handle two ESR releases when there's currently only one. Other than setting `channel_esr_next = channel_esr` here locally to fake it, is there any other more reliable way?

Reply:
I don't really think so. Faking that data is kind of involved since it's updating the `firefox_versions.json` with a version in that key and making sure that the version you used has builds in `firefox_primary_builds.json`. So just setting your variable here to use other builds to fake it sounds like the easiest way to go.